### PR TITLE
bpf2go: write dependencies to temporary file to support Windows

### PR DESCRIPTION
**Pull request by junjiexing**
I want to develop Android eBPF programs on Windows, but found that bpf2go uses `/dev/fd/3` to accept dependencies output, which is not supported on Windows. This commit modified it to use temporary files to accept dependencies.

### CODE REVIEW

File: cmd/bpf2go/compile.go [hunk 1]
```diff
@@ -70,45 +70,35 @@ func compile(args compileArgs) error {
 	)
 	cmd.Dir = args.dir
 
-	var depRd, depWr *os.File
+	var depFile *os.File
 	if args.dep != nil {
-		depRd, depWr, err = os.Pipe()
+		depFile, err = os.CreateTemp(os.TempDir(), "bpf2go")
```
Comment (lmb) [inline cmd/bpf2go/compile.go:75]:
```suggestion
		depFile, err = os.CreateTemp("", "bpf2go")
```

**Review by lmb — APPROVED** [review]
One small nit. Please be aware that this doesn't mean we won't break Windows support going forward, since none of us use Windows to develop and we don't use it in CI.

File: cmd/bpf2go/compile.go [hunk 2]
```diff
@@ -70,45 +70,35 @@ func compile(args compileArgs) error {
...
+	var depFile *os.File
 	if args.dep != nil {
-		depRd, depWr, err = os.Pipe()
+		depFile, err = os.CreateTemp("", "bpf2go")
 		if err != nil {
 			return err
 		}
-		defer depRd.Close()
-		defer depWr.Close()
+		defer os.Remove(depFile.Name())
```
Comment (ti-mo) [inline cmd/bpf2go/compile.go:79]: This is missing a `defer depFile.Close()`.

**Review by ti-mo — DISMISSED** [review]

**Comment by lmb** [comment]
Thanks!

**Pull request merged**